ArcGISRuntimeSDKQt_CppSamples/Layers/IdentifyRasterCell/IdentifyRasterCell.cpp

```diff
@@ -0,0 +1,172 @@
+#else
+  dataPath = QDir::homePath();
+#endif
+
+  return dataPath;
+}
+} // namespace
+
+IdentifyRasterCell::IdentifyRasterCell(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_map(new Map(Basemap::oceans(this), this))
+{
+  // initialize the raster layer
+  const QString filepath = defaultDataPath() + "/ArcGIS/Runtime/Data/raster/SA_EVI_8Day_03May20/SA_EVI_8Day_03May20.tif";
+  Raster* raster = new Raster(filepath, this);
+  m_rasterLayer = new RasterLayer(raster, this);
+
+  m_map->operationalLayers()->append(m_rasterLayer);
+}
+
+IdentifyRasterCell::~IdentifyRasterCell() = default;
+
+void IdentifyRasterCell::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<IdentifyRasterCell>("Esri.Samples", 1, 0, "IdentifyRasterCellSample");
+}
+
+MapQuickView* IdentifyRasterCell::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void IdentifyRasterCell::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  connectSignals();
+
+  emit mapViewChanged();
+}
+
+void IdentifyRasterCell::connectSignals()
+{
+  connect(m_rasterLayer, &Layer::doneLoading, this, [this](Error error)
+  {
+    if (!error.isEmpty())
+    {
+      qDebug() << error.message() << error.additionalMessage();
+      return;
+    }
+
+    m_mapView->setViewpointGeometry(m_rasterLayer->fullExtent());
+  });
+
+  connect(m_mapView, &MapQuickView::identifyLayerCompleted, this, [this](QUuid, IdentifyLayerResult* rawIdentifyResult)
+  {
+    auto identifyResult = std::unique_ptr<IdentifyLayerResult>(rawIdentifyResult);
+
+    for (GeoElement* geoElement : identifyResult->geoElements())
+    {
+      if (RasterCell* rasterCell = dynamic_cast<RasterCell*>(geoElement))
+      {
+        QString calloutString;
+        const AttributeListModel* attributes = rasterCell->attributes();
+        const QStringList attributeNames = rasterCell->attributes()->attributeNames();
+
+        for (int i = 0; i < attributeNames.size(); ++i)
+        {
+          const QString value = QVariant((*attributes)[attributeNames[i]]).toString();
+          calloutString.append(attributeNames[i] + ": " + value + "\n");
+        }
+
+        const double xPoint = rasterCell->geometry().extent().xMin();
+        const double yPoint = rasterCell->geometry().extent().yMin();
+
+        calloutString.append("X: " + QString::number(xPoint, 'f', 2) + " Y: " + QString::number(yPoint, 'f', 2));
+
+        m_mapView->calloutData()->setLocation(m_clickedPoint);
+        m_mapView->calloutData()->setDetail(calloutString);
+        m_mapView->calloutData()->setVisible(true);
+        m_calloutData = m_mapView->calloutData();
+        emit calloutDataChanged();
+      }
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](const QMouseEvent& e)
```

Comment:
do we want to identify on both `mouseClicked` and `mousePressedAndHeld`? Seems like we should use one or the other